Scripts/freight_lem.py

```diff
@@ -0,0 +1,161 @@
+            ass_model._create_matrix(matrix_id, matrix_name, matrix_type, overwrite=True)
+            matrix_counter += 1
+            ass_model.freight_network.emme_matrices.update(
+                {matrix_name: {matrix_type: matrix_id}})
+            ass_model.freight_network.set_matrix(matrix_name, demand[mode])
+            with resultmatrices.open("freight_demand", "vrk", zone_numbers, m="a") as mtx:
+                mtx[matrix_name] = demand[mode]
+        if purpose in args.specify_commodity_names:
+            ass_model.freight_network.save_network_volumes(purpose)
+            matrix_counter += 7 # Shift to next 10 digit
+        ass_model.freight_network.output_traversal_matrix(resultdata.path)
+        aux_demand = transform_traversal_data(resultdata.path, zone_numbers)
+    log.info("Simulation ready.")
+
+if __name__ == "__main__":
+    parser = ArgumentParser(epilog="Freight lem-model-system entry point script.")
+    config = utils.config.read_from_file()
+    config["SAVE_EMME_MATRICES"] = True
+    config["SPECIFY_COMMODITY_NAMES"] = ["marita"]
+    
+    parser.add_argument(
+        "--log-level",
+        choices={"DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"})
+    parser.add_argument(
+        "--log-format",
+        choices={"TEXT", "JSON"})
+    parser.add_argument(
+        "--scenario-name",
+        type=str,
+        help="Scenario name"),
+    parser.add_argument(
+        "--baseline-data-path",
+        type=str,
+        help="Path to folder containing both baseline zonedata and -matrices"),
+    parser.add_argument(
+        "--cost-data-path",
+        type=str,
+        help="Path to file containing transport cost data"),
+    parser.add_argument(
+        "--results-path",
+        type=str,
+        help="Path to folder where result data is saved to."),
+    parser.add_argument(
+        "--emme-path",
+        type=str,
+        help="Filepath to .emp EMME-project-file"),
+    parser.add_argument(
+        "--first-scenario-id",
+        type=int,
+        help="First (biking) scenario ID within EMME project (.emp)."),
+    parser.add_argument(
+        "--save-emme-matrices",
+        type=bool,
+        help="Using this flag saves matrices for specified commodities."),
+    parser.add_argument(
+        "--first-matrix-id",
+        type=int,
+        help="First matrix ID within EMME project (.emp)."),
+    parser.add_argument(
+        "--specify-commodity-names",
+        type=List[str],
+        help="Commodity names in 29 classification. Assigned and saved as mtx. Currently max 3."),
+
+    parser.set_defaults(
+        **{key.lower(): val for key, val in config.items()})
+    args = parser.parse_args()
+    
+    if not isinstance(args.specify_commodity_names, list):
+        raise TypeError("Invalid type for save commodity names")
+    for name in args.specify_commodity_names:
+        if name not in list(commodity_conversion):
+            raise ValueError("Invalid given commodity name.")
```

Comment:
I think that what you want to use is the `nargs="*"` and `choices=commodity_conversion` arguments in `add_argument` for "--specify-commodity-names". Setting `type` is unnecessary when dealing with strings. See docs here: https://docs.python.org/3/library/argparse.html#nargs